src/java/voldemort/store/readonly/swapper/AdminStoreSwapper.java

```diff
@@ -282,6 +307,10 @@ private String fetch(String storeDir) {
                     errorMessage = "Exception during push. Swap will be aborted.";
                 }
                 throw new VoldemortException(errorMessage);
+
+            }*/
+            if (!swapIsPossible){
+                throw new VoldemortException("Exception during psh. Swap will be aborted", fetchResponseMap.get(failedNodes.get(0)).getException());
```

Comment:
Can we mention which node gave us this exception?

For example: You could add this to the end of the error message string: ```"Exception during push. Swap will be aborted. Re-throwing exception from " + failedNodes.get(0).briefToString()```.